render.pl

```diff
@@ -48,28 +64,47 @@
 $safe_content =~ s/\.\s+(?=\.)/./g; # turn . . . into ...
 
 if(!$test) {
+  print "---- Generating silence and sound effect mp3 files...\n";
   #create silence
-  system('rm -f silence.mp3');
-  system("ffmpeg -f lavfi -i anullsrc=channel_layout=5.1:sample_rate=22050 -t $silence_between_sets -codec:a libmp3lame -b:a 256k silence.mp3");
+  unlink "silence.mp3" if (-f "silence.mp3");
+  @cmdLst = ("ffmpeg", "-f", "lavfi", "-i", "anullsrc=channel_layout=5.1:sample_rate=22050", "-t",
+             "$silence_between_sets", "-codec:a", "libmp3lame", "-b:a", "256k", "silence.mp3");
+  # print("cmd-1: @cmdLst\n");
+  system(@cmdLst) == 0 or die "ERROR 1: @cmdLst failed, $!\n";
 
   # This is the silence between the Morse code and the spoken voice
-  system('rm -f silence1.mp3');
-  system("ffmpeg -f lavfi -i anullsrc=channel_layout=5.1:sample_rate=22050 -t $silence_between_morse_code_and_spoken_voice -codec:a libmp3lame -b:a 256k silence1.mp3");
+  unlink "silence1.mp3" if (-f "silence1.mp3");
+  @cmdLst = ("ffmpeg", "-f", "lavfi", "-i", "anullsrc=channel_layout=5.1:sample_rate=22050",
+             "-t", "$silence_between_morse_code_and_spoken_voice", "-codec:a", "libmp3lame",
+             "-b:a", "256k", "silence1.mp3");
+  # print "cmd-2: @cmdLst\n";
+  system(@cmdLst) == 0 or die "ERROR 2: @cmdLst failed, $!\n";
 
   # This is the silence between the Morse code and the spoken voice
-  system('rm -f silence2.mp3');
-  system("ffmpeg -f lavfi -i anullsrc=channel_layout=5.1:sample_rate=22050 -t $silence_between_voice_and_repeat -codec:a libmp3lame -b:a 256k silence2.mp3");
+  unlink 'silence2.mp3' if (-f 'silence2.mp3');
+  @cmdLst = ("ffmpeg", "-f", "lavfi", "-i", "anullsrc=channel_layout=5.1:sample_rate=22050",
+             "-t", "$silence_between_voice_and_repeat", "-codec:a", "libmp3lame",
+             "-b:a", "256k", "silence2.mp3");
+  # print "cmd-3: @cmdLst\n";
+  system(@cmdLst) == 0 or die "ERROR 3: @cmdLst failed, $!\n";
 
   #create quieter tone
-  system('rm -f plink-softer.mp3');
-  system('ffmpeg -i sounds/plink.mp3 -filter:a "volume=0.5" plink-softer.mp3');
+  unlink 'plink-softer.mp3' if (-f 'plink-softer.mp3');
+  $cmd = 'ffmpeg -i sounds/plink.mp3 -filter:a "volume=0.5" plink-softer.mp3';
+  # print "cmd-4: $cmd\n";
+  system($cmd) == 0 or die "ERROR 4: $cmd failed, $!\n";
 
   #create quieter tone
-  system('rm -f pluck-softer.mp3');
-  system('ffmpeg -i sounds/pluck.mp3 -filter:a "volume=0.5" pluck-softer.mp3');
-
-  # make sure the cache directory exists
-  system('mkdir cache >& /dev/null');
+  unlink 'pluck-softer.mp3' if (-f 'pluck-softer.mp3');
+  $cmd = 'ffmpeg -i sounds/pluck.mp3 -filter:a "volume=0.5" pluck-softer.mp3';
+  # print "cmd-5: $cmd\n";
+  system($cmd) == 0 or die "ERROR 5: $cmd failed, $!\n";
+
+  if (-d "cache") {
```

Comment:
Hmm.. I'm not sure it is a great idea to delete the cache directory on every invocation. If you are worried that someone might change from "neural" to "standard" and forget to delete the cache directory, it might be better to change the hash so that it is the text-2-speech configuration + the text to be spoken. It can save a lot of money by not deleting it unless I have a reason to.